osrelease/scanner.go

```diff
@@ -123,21 +139,48 @@ func parse(ctx context.Context, r io.Reader) (*claircore.Distribution, error) {
 
 		switch key {
 		case "ID":
+			log.Debug().Msg("found ID")
 			d.DID = value
 		case "VERSION_ID":
+			log.Debug().Msg("found VERSION_ID")
 			d.VersionID = value
 		case "BUILD_ID":
 		case "VARIANT_ID":
 		case "CPE_NAME":
-			d.CPE = value
+			log.Debug().Msg("found CPE_NAME")
+			var err error
+			var wfn common.WellFormedName
+			switch {
+			case common.ValidateURI(value) == nil:
+				wfn, err = naming.UnbindURI(value)
+			case common.ValidateFS(value) == nil:
+				wfn, err = naming.UnbindFS(value)
+			}
+			if err != nil {
+				log.Warn().
+					Err(err).
+					Str("value", value).
+					Msg("failed to unbind the cpe")
+				break
+			}
+			d.CPE = naming.BindToURI(wfn)
 		case "NAME":
+			log.Debug().Msg("found NAME")
 			d.Name = value
 		case "VERSION":
+			log.Debug().Msg("found VERSION")
 			d.Version = value
 		case "ID_LIKE":
 		case "VERSION_CODENAME":
+			log.Debug().Msg("found VERISON_CODENAME")
 			d.VersionCodeName = value
 		case "PRETTY_NAME":
+			log.Debug().Msg("found PRETTY_NAME")
+			d.PrettyName = value
+		case "REDHAT_BUGZILLA_PRODUCT":
```

Comment:
I would feel better if this was done correctly and we plumbed it into the datamodel. If it's a field that's worth having on the os-release file we don't need to munge it.

Reply:
Lets void this for now. I think this becomes a moot point once we introduce inverted index.

Reply:
you mean by loading the os-release file into something like an hstore? That's currently my thinking on how we should approach this.

Reply:
@hdonnay right now the inverted_index branch has us just injesting the vulnerabilities as jsonb documents to obtain the same thing as Hstore in principal. You can make a GIN index on hstore also. But I implemented as jsonb so far.